enum_methods.rb

```diff
@@ -0,0 +1,164 @@
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times do |i|
+      yield(to_a[i])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times do |i|
+      yield(to_a[i], i)
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  def my_all2?
+    return to_enum unless block_given?
+
```

Comment:
* [ ] There should be a working `my_all?`  identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
``` p [ ].my_all?``` != ```p [ ].all?```

Reply:
The other one is the correct, I forgot to remove this!